raiden/network/matrixtransport.py

```diff
@@ -178,15 +195,27 @@ def send_async(
         return self._messageids_to_asyncresult[message_id]
 
     def stop_and_wait(self):
-        self._client.set_presence_state(UserPresence.OFFLINE.value)
-        self._client.stop_listener_thread()
-        self._client.logout()
+        if self._running:
+            self._running = False
+            self._client.set_presence_state(UserPresence.OFFLINE.value)
+            self._client.stop_listener_thread()
+
+            # Set all the pending results to False, this will also
+            # cause pending retries to be aborted
+            for async_result in self._messageids_to_asyncresult.values():
+                async_result.set(False)
 
-        # Set all the pending results to False, this will also cause pending retries to be aborted
-        for async_result in self._messageids_to_asyncresult.values():
-            async_result.set(False)
+            gevent.wait(self.greenlets)
+            self._client.logout()
 
-        gevent.wait(self.greenlets)
+    @property
+    def log(self):
+        if self._bound_logger:
+            return self._bound_logger
+        if not getattr(self._client, 'user_id', None):
+            return log
+        self._bound_logger = log.bind(current_user=self._client.user_id)
```

Comment:
is this the node id? can we please add it or renamed it?

Reply:
@hackaugusto no, this is the matrix userId, as in `@0x<eth_addr>.<rand_sufix>:transport.raiden.network`